stack_data/core.py

```diff
@@ -734,6 +751,16 @@ def lines(self) -> List[Union[Line, LineGap]]:
         for line in real_lines:
             line.leading_indent = leading_indent
 
+        if not self.skip_blank:
+            new_lines = [result[0]]
```

Comment:
I was thinking of doing that, but I thought it would give a single piece of code with too convoluted logic. By separating it out, I thought it would be easier to understand. I'll try to fold it in. However, with the third case you want me to add, it might get confusing to a reader.